Use library_code_facet_ssim for the library

jcoyne opened the pull request:
This allows us to change the label with config, rather than have to reindex everything.  Before we activate this facet in the new UI, we need to populate it in the indexer.
<!-- Closes #ISSUE_NUMBER -->
<!-- 📝 CHANGELOG update? -->
Ref #4791

See https://github.com/sul-dlss/searchworks_traject_indexer/pull/1638

cbeer commented:
I think we're ready to review this in light of #5210. Can you rebase this PR?

Two comments to start:
- I wonder if there's a way to keep existing urls with `building_facet` working (in case it's been bookmarked or linked to externally), perhaps by translating it to the `library_code_facet_ssim` behind the scenes?

- #5210 calls for having multiple library codes show the same label. Do you think this approach is compatible with that easily (say, but  non-library code like SDR and having things in SAL1-2/SAL-NEWARK/SAL3 use that for indexing?), or is there something better?

jcoyne commented:
@cbeer I rebased and added the redirect from the old facet values. I think the "Off-campus collections" first requires us to index all the SALs together as one code.
